uldk_gugik_dialog.py

```diff
@@ -23,154 +23,95 @@
+                comboboxes_to_hide = list(list(RADIOBUTTON_COMBOBOX_MAPPING.values())[combo_idx:])
+                break
+        for combo in comboboxes_to_hide:
+            combo_obj = getattr(self, combo)
+            combo_obj.setStyleSheet("QComboBox { color: transparent }")
+            combo_obj.setEnabled(False)
+
+    def setup_administrative_unit_obj(self, func, dependent_combo):
+        combo_obj = getattr(self, dependent_combo)
+        unit_data = self.sender().currentData()
+        if not unit_data:
+            return
+        combo_obj.clear()
+        unit_dict = getattr(self.regionFetch, func)(unit_data)
+        combo_obj.addItems(unit_dict.values())
+        for idx, val in enumerate(unit_dict.keys()):
+            combo_obj.setItemData(idx, val)
+        combo_obj.setCurrentIndex(-1)
 
     def closeEvent(self, event):
         self.closingPlugin.emit()
         event.accept()
-
-    def wojcomboBox_currentTextChanged(self, text):
-        self.powcomboBox.clear()
-        if not self.rdb_pw.isChecked():
-            self.handleResponseVoivode(False)
-            self.ctrl_rest_obj(False)
-
-        self.powiatDictionary = self.regionFetch.getPowiatDictByWojewodztwoName(text)
-        data = {k: v[0] for k, v in self.powiatDictionary.items()}
-        self.powcomboBox.addItems(list(self.powiatDictionary.keys()))
-
-        for idx, po in enumerate(data.keys()):
-            self.powcomboBox.setItemData(idx, data[po])
-
-        self.handleResponseVoivode(True)
-        self.ctrl_rest_obj(True)        
-
-    def powcomboBox_currentTextChanged(self, text):
-        self.gmicomboBox.clear()
-        if not self.rdb_pw.isChecked():
-            self.handleResponseCounty(False)
-            self.ctrl_rest_obj(False)
-
-        self.gminaDictionary = self.regionFetch.getGminaDictByPowiatName(text)
-        data = {k: v[0] for k, v in self.gminaDictionary.items()}
-        self.gmicomboBox.addItems(list(self.gminaDictionary.keys()))
-
-        for idx, po in enumerate(data.keys()):
-            self.gmicomboBox.setItemData(idx, data[po])
-
-        self.handleResponseCounty(True)
-        self.ctrl_rest_obj(True)
-
-    def gmicomboBox_currentTextChanged(self, text):
-        self.obrcomboBox.clear()
-        if self.wojcomboBox.currentTextChanged or self.powcomboBox.currentTextChanged:
-            if self.rdb_gm.isChecked() or self.rdb_ob.isChecked() or self.rdb_dz.isChecked():
-                self.handleResponseMunicip(False)
-                self.ctrl_rest_obj(False)
-
-
-        self.obrebDictionary = self.regionFetch.getObrebDictByGminaName(text)
-        data = {k: v for k, v in self.obrebDictionary.items()}
-        self.obrcomboBox.addItems(list(self.obrebDictionary.keys()))
-
-        for idx, ob in enumerate(data.keys()):
-            self.obrcomboBox.setItemData(idx, data[ob])
-
-        self.handleResponseMunicip(True)
-        self.ctrl_rest_obj(True)
```

Comment:
Brakuje mi w aktualnej implementacji funkcji ctrl_rest_obj lub podobnej do niej. Teraz można pobrać działki i inne obiekty nawet jak nie zostaną poprawnie ściągnięte z serwera. Jest to mylące, więc trzeba by przywrócić to 1:1 lub przywrócić i może podrasować.